rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/ParametricTextDocumentService.java

```diff
@@ -472,14 +491,15 @@ public CompletableFuture<SemanticTokens> semanticTokensRange(SemanticTokensRange
         final TextDocumentState file = getFile(params.getTextDocument());
         ILanguageContributions contrib = contributions(params.getTextDocument());
         return recoverExceptions(file.getCurrentTreeAsync()
+            .thenApply(Versioned::get)
             .thenApply(contrib::outline)
             .thenCompose(InterruptibleFuture::get)
             .thenApply(c -> Outline.buildOutline(c, columns.get(file.getLocation())))
             , Collections::emptyList);
     }
 
     private ParametricSummaryBridge summary(TextDocumentIdentifier doc) {
```

Comment:
have you reviewed callers? as the `summary` function was normally also called to dynamically invoke the builder when needed. and we want to have goto-definition etc still work, even without saving. if you have an analyzer configured.